ariadne/tracknet_v2_1/processor.py

```diff
@@ -81,72 +76,77 @@ def preprocess_chunk(self,
                            chunks: List[ProcessedTracknetDataChunk]) -> ProcessedTracknetData:
         for chunk in chunks:
+            if chunk.processed_object is None or len(chunk.processed_object)==0:
+                continue
             chunk_data_x = []
             chunk_data_y = []
             chunk_data_len = []
             chunk_data_real = []
             chunk_data_moment = []
             df = chunk.processed_object
+
             grouped_df = df[df['track'] != -1].groupby('track')
+            last_station = df[df['station'] > 1][['phi', 'z']].values
+            if grouped_df.ngroups == 0:
+                print('multiplicity is 0!')
+                chunk.processed_object = None
+                continue
             for i, data in grouped_df:
                 chunk_data_x.append(data[['r', 'phi', 'z']].values[:-1])
-                chunk_data_y.append(data[['r', 'phi', 'z']].values[-1])
+                chunk_data_y.append(data[['phi', 'z']].values[-1])
                 chunk_data_len.append(2)
                 chunk_data_moment.append(data[['px','py','pz']].values[-1])
                 chunk_data_real.append(1)
             print('=====> id', chunk.id)
+            multiplicity = len(chunk_data_x)
             first_station = df[df['station'] == 0][['r', 'phi', 'z', 'track']]
             first_station.columns = ['r_left', 'phi_left', 'z_left', 'track_left']
 
             second_station = df[df['station'] == 1][['r', 'phi', 'z', 'track']]
             second_station.columns = ['r_right', 'phi_right', 'z_right', 'track_right']
 
-            fake_tracks = self.cartesian(first_station, second_station)
-            fake_tracks = fake_tracks.sample(n=int(fake_tracks.shape[0] / 10), random_state=1)
+            fake_tracks = cartesian(first_station, second_station)
+            fake_tracks = fake_tracks.sample(n=int(fake_tracks.shape[0] / 2), random_state=1)
             for i, row in tqdm(fake_tracks.iterrows()):
                 temp_data = np.zeros((2, 3))
                 temp_data[0, :] = row[['r_left', 'phi_left', 'z_left']].values
                 temp_data[1, :] = row[['r_right', 'phi_right', 'z_right']].values
+
                 chunk_data_x.append(temp_data)
                 #print(chunk_data_x[-1].shape)
                 chunk_data_y.append(chunk_data_y[0])
                 chunk_data_moment.append(chunk_data_moment[0])
                 chunk_data_real.append(0)
                 chunk_data_len.append(2)
-
             chunk_data_x = np.stack(chunk_data_x, axis=0)
             chunk_data_y = np.stack(chunk_data_y, axis=0)
             chunk_data_moment = np.stack(chunk_data_moment, axis=0)
             chunk_data_real = np.stack(chunk_data_real, axis=0)
             chunk_data_len = np.stack(chunk_data_len, axis=0)
             chunk_data_event = np.ones(chunk_data_len.shape)
 
+            chunk_data_event_last_station = np.ones(last_station.shape[0])
+            chunk_data_event_last_station *= chunk.id
```

Comment:
if you will use the code above, this becomes redundant

Reply:
replaced all with np.full